panic: runtime error: invalid memory address or nil pointer dereference

### Terraform Version
```
v0.1.4
```
required_providers 
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 2.73.0"
    }
(Note: The same error for 2.74 and 2.75!)
### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For Terraform configs larger than a few resources, or that involve multiple files, please make a GitHub repository that we can clone, rather than copy-pasting multiple files in here. For security, you can also encrypt the files using our GPG public key at https://www.hashicorp.com/security.
-->
File1 (part of it)
```
locals {
  publisher_name    = "xx"
  publisher_email   = xx@xx.com"
  api_name          = "xxi"
  api_operation_id  = "xx"
  endpoint_url      = "/xx"
}

resource "azurerm_api_management" "this" {
  count               = var.deploy_to_vnet ? 1 : 0
  name                = local.apm_name
  resource_group_name = data.azurerm_resource_group.this.name
  location            = var.location  
  publisher_name      = local.publisher_name
  publisher_email     = local.publisher_email
  tags = {
    environment = "${var.env_code}"
  }

  sku_name            = "Standard_1"
}
```
File 2(part of it)
```
resource "azurerm_linux_virtual_machine_scale_set" "this" {
  count                           = var.deploy_to_vnet ? 1 : 0  
  name                            = local.vmss_name
  resource_group_name             = data.azurerm_resource_group.this.name
  location                        = var.location
  sku                             = "Standard_DS1_v2"
  instances                       = 0
  admin_username                  = var.vmss_admin_user
  admin_password                  = random_password.password.result
  disable_password_authentication = false

  source_image_id                 = var.source_image_id
  
  network_interface {
    name    = "example"
    primary = true

    ip_configuration {
      name      = "internal"
      primary   = true
      subnet_id = azurerm_subnet.aml[count.index].id
    }
  }

  os_disk {
    storage_account_type = "Standard_LRS"
    caching              = "ReadWrite"
  }

  identity {
    type = "SystemAssigned"
  }

  lifecycle {
    ignore_changes = [
      instances,
      tags
    ]
  }

  tags = {
    environment = "${var.env_code}"
  }

  depends_on = [
    azurerm_virtual_network.this
  ]

}
```

### Debug Output

│ Error: Plugin did not respond
│ 
│   with azurerm_linux_virtual_machine_scale_set.this[0],
│   on vmss.tf line 1, in resource "azurerm_linux_virtual_machine_scale_set" "this":
│    1: resource "azurerm_linux_virtual_machine_scale_set" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵


│ Error: Plugin did not respond
│ 
│   with azurerm_api_management.this[0],
│   on apim.tf line 9, in resource "azurerm_api_management" "this":
│    9: resource "azurerm_api_management" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.


### Crash Output

Stack trace from the terraform-provider-azurerm_v2.73.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4952260]

goroutine 120 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.resourceLinuxVirtualMachineScaleSetUpdate(0xc001cc5d00, 0x588e0c0, 0xc000f30700, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go:622 +0x33a0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc0007ec380, 0x6844cc8, 0xc00124aac0, 0xc001cc5d00, 0x588e0c0, 0xc000f30700, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:352 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0007ec380, 0x6844cc8, 0xc00124aac0, 0xc001cea1c0, 0xc0009b89e0, 0x588e0c0, 0xc000f30700, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0010b8528, 0x6844cc8, 0xc00124aac0, 0xc0016cd950, 0xc00124aac0, 0x5d7a940, 0xc001738800)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc001b4b0e0, 0x6844d70, 0xc00124aac0, 0xc001cea070, 0xc001b4b0e0, 0xc001738810, 0xc001706ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x5d7a940, 0xc001b4b0e0, 0x6844d70, 0xc001738810, 0xc001f510e0, 0x0, 0x6844d70, 0xc001738810, 0xc0018c4500, 0x2011)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00022e8c0, 0x6892598, 0xc001109680, 0xc0008db0e0, 0xc0014d50b0, 0x9e56140, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00022e8c0, 0x6892598, 0xc001109680, 0xc0008db0e0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0009f6ff0, 0xc00022e8c0, 0x6892598, 0xc001109680, 0xc0008db0e0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd

Error: The terraform-provider-azurerm_v2.73.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.


### Expected Behavior
VMSS should be updated
Azure API Management should be newly created

### Actual Behavior
Error see above

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

--- a/internal/services/compute/linux_virtual_machine_scale_set_resource.go
+++ b/internal/services/compute/linux_virtual_machine_scale_set_resource.go
@@ -617,9 +617,9 @@ func resourceLinuxVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData, meta i
 			automaticRaw := d.Get("automatic_os_upgrade_policy").([]interface{})
 			upgradePolicy.AutomaticOSUpgradePolicy = ExpandVirtualMachineScaleSetAutomaticUpgradePolicy(automaticRaw)
 
-			// however if this block has been changed then we need to pull it
-			// we can guarantee this always has a value since it'll have been expanded and thus is safe to de-ref
-			automaticOSUpgradeIsEnabled = *upgradePolicy.AutomaticOSUpgradePolicy.EnableAutomaticOSUpgrade
+			if upgradePolicy.AutomaticOSUpgradePolicy != nil {
+				automaticOSUpgradeIsEnabled = *upgradePolicy.AutomaticOSUpgradePolicy.EnableAutomaticOSUpgrade
+			}
 		}
 
 		if d.HasChange("rolling_upgrade_policy") {

